src/workerd/api/pyodide/pyodide.c++

```diff
@@ -421,6 +421,16 @@ kj::Own<PyodideMetadataReader::State> PyodideMetadataReader::State::clone() {
   return kj::heap<PyodideMetadataReader::State>(*this);
 }
 
+void PyodideMetadataReader::State::verifyNoMainModuleInVendor() {
+  // Verify that we don't have module named after the main module in the `vendor` subdir.
+  for (auto& name: moduleInfo.names) {
+    if (name.startsWith(kj::str("vendor/", mainModule))) {
+      JSG_FAIL_REQUIRE(
+          Error, kj::str("Python module vendor/", mainModule, " clashes with main module"));
+    }
+  }
```

Comment:
If the main module is called `blah.py` they could still have `vendor/blah/__init__.py` and that would also shadow. Or `vendor/blah.so`. I don't know if we care about these cases.

Reply:
Good point, I think we do care about these too.